pkg/secrets/secrets.go

```diff
@@ -170,17 +180,23 @@ func (s *SecretsManager) Store(name string, data []byte, driverType string, driv
 		}
 	}
 
-	if metadata == nil {
-		metadata = make(map[string]string)
+	if options.Metadata == nil {
+		options.Metadata = make(map[string]string)
+	}
+	if options.Labels == nil {
+		options.Labels = make(map[string]string)
+	}
+	if options.DriverOpts == nil {
+		options.DriverOpts = make(map[string]string)
```

Comment:
Is there any specific reason why they cannot be nil? If these are optional I always prefer nil pointers vs empty maps.

Reply:
It's either we check here and store empty maps, or later on, when getting a secret, checking if the maps != nil each time. I feel like storing empty maps are a little less error prone, but thats also personal preference.